### ISSUE TITLE
"And" method not working as expected

### ISSUE THREAD

**Issue by jaandrews** [Shazwazza/Examine#137]
Ref: [umbraco/Umbraco-CMS#6316](https://github.com/umbraco/Umbraco-CMS/issues/6316)

The issue is that the And() call doesn't seem to do anything. I have a query where the default operation is set to "BooleanOperation.Or." The first half of the query is a group created using the following code.
```csharp
if (config.LimitToParents != null && config.LimitToParents.Any()) {
    result = criteria.Group(group =>
    {
        //var innerQuery = group.GroupedOr(new[] {"parentID"},
        //    config.LimitToParents.Select(id => id.ToString()).ToArray());

        var first = config.LimitToParents.First();
        var innerQuery = group.Field("parentID", first);
        foreach (var parentId in config.LimitToParents.Skip(1))
        {
            innerQuery = innerQuery.Or().Field("parentID", parentId);
        }
        return innerQuery;
    });
}
```
"LimitToParents" is an array of parent ids. The second half filters on a "categories" property that stores a comma separated node keys with their dashes removed. An example of the resulting query for this portion is
```csharp
categories:a3bf27740d0741528824ca6e2cefd81b
```
However, when I attempt to join these two operations with "And()" I get something like
```csharp
(parentID[1234 TO 1234] parentID[1235 TO 1235]) (categories:a3bf27740d0741528824ca6e2cefd81b)
```
instead of
```csharp
+(parentID[1234 TO 1234] parentID[1235 TO 1235]) +(categories:a3bf27740d0741528824ca6e2cefd81b)
```
so the results include items that fulfill only one of the desired conditions. Am I misunderstanding what "And()" does or is this a bug?

**Comment by Shazwazza** [comment]
Hi, thanks for reporting, i will need to write a unit test to check all of this. If you have any time it would be wonderful to have a unit test written to prove or disprove this as a PR?

**Comment by Shazwazza** [comment]
I found an issue that might just be the issue you are seeing with sub queries. 

If you have a look at this test: https://github.com/Shazwazza/Examine/blob/master/src/Examine.Test/Search/FluentApiTests.cs#L1615

The output query is now: `(+Type:type1 +(Content:world Content:something)) (+Type:type2 +(+Content:world +Content:cruel))` which is what we want but before the fix that I've pushed, this was ending up like: `+(Type:type1 +(Content:world Content:something)) +(Type:type2 +(+Content:world +Content:cruel))`

I'll make a new release and an issue to track that one then you can test to see if it solves your issue. Will post back here when it's up on nuget

**Comment by Shazwazza** [comment]
Have published a new version: https://github.com/Shazwazza/Examine/releases/tag/v1.0.2

**Comment by jaandrews** [comment]
Thanks for the update Shannon. I'll give the update a try and let you know how it goes.

**Comment by Shazwazza** [comment]
Closing this issue due to inactivity, assuming it's fixed with 1.0.2
